### PR TITLE
expenses plugin redesign

### PR DESCRIPTION
**Pull request by m1well**
after a discussion with @dwertheimer I created a whole redesign of my expenses plugin.
some of the changes:
* columorder is configurable
* delimiter is configurable
* 3 more commands for faster usage
* more exception handling
* more tests
* more type checks on some areas
* ...

### CODE REVIEW

File: helpers/general.js [hunk 1]
```diff
@@ -169,15 +169,3 @@ export function capitalize(s: string): string {
   if (typeof s !== 'string') return ''
   return s.charAt(0).toUpperCase() + s.slice(1)
 }
-
-/**
```
Comment (dwertheimer) [inline helpers/general.js:173]: did you mean to move this out of helpers/general? normally you would want to do this as a separate PR since this is a global helper file. Probably ok because nobody else seems to have used it since you added it. But why did you move it?

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,4 @@ package-lock.json
 lcov*
 coverage/
 .history/*.*
+xtemp/
```
Comment (dwertheimer) [inline .gitignore:20]: what's xtemp?

**Review by dwertheimer — APPROVED** [review]
Looks really great. I tested it and works! Thanks!

File: .gitignore (hunk 2 above)
Comment (m1well) [reply .gitignore:20]: just a folder where you can put some files which aren't tracked via git.

e.g.:
* sometimes I just copy my "real expenses" to my IDE to have a backup if I am testing in Noteplan.
* or I had to rewrite all `2021;02` dates to `2021-02` and I did that in my IDE

File: helpers/general.js (hunk 1 above)
Comment (m1well) [reply helpers/general.js:173]: unfortunately i had to move this function because my tests did not work anymore.
in this `general` there is a reference to `DataStore` and then the test always breaks, because I don't know how to mock the `DataStore`.
Of course I don't move functions which are in use by anybody else ;)

File: .gitignore (hunk 2 above)
Comment (dwertheimer) [reply .gitignore:20]: feels like everyone else doesn't need this though. can you do this locally and discard the change from the PR?

File: helpers/general.js (hunk 1 above)
Comment (dwertheimer) [reply helpers/general.js:173]: got it. makes sense.

File: .gitignore (hunk 2 above)
Comment (m1well) [reply .gitignore:20]: ok :D
